Documenting the usage of return type for programmatic navigation (#1480)
vaadin/docs: articles/routing/route-parameters.asciidoc

Note that more specific route always takes precedence over wildcard route.
Note also that if no value was passed to the wildcard route parameter, then its value will be an empty String.

> == Maintaining Route Parameters manually

Reviewer: ```suggestion
== Maintaining Route Parameters Manually
```

> == Maintaining Route Parameters Manually

Reviewer 2: First, I think this is a separate topic from documenting programmatic navigation. So let's create a separate PR for it. 

That aside, I must say that I read this section three times, and I'm still struggling to understand the solution it's describing or even the problem that it's intended to solve. But as far as I can tell, this is an esoteric use case that should be covered in an "advanced section" somewhere (I don't think it's suitable for people who are just starting to learn about how to add parameters to a Vaadin route). I suggest either creating a new article for that under `routing/additional-guides` or adding it to the History API guide.

Author: The example could indeed be better. Think of a search view where you want to maintain the deep linking URL in the browser. You don't want to use our somewhat clumsy routing/navigating API to navigate to a new search view with new parameters. Will be much better with this approach (or with query parameters).

Author: Changed the title and added a bit more context why using Router for everything isn't the best way to make the world a better place.

> In certain cases, developer might want to maintain the parameters in the browser URL. 
> Like if the edited entity in a form changes via instance method instead of via passing URL parameters.
> In this case developer can use the History object to maintain the RouteParameters.

Reviewer 3: ```suggestion
In certain cases, a developer might want to maintain the parameters in the browser URL, such as if the edited entity in a form changes via an instance method instead of via passing URL parameters.
In this case, the developer can use the [classname]`History` object to maintain the [classname]`RouteParameters`.
```

> *Example*: A view that is updating the deep linking parameters based on a parameter given in a public API. The view also implements HasUrlParameter to de-serialise the parameter in case a user arrives directly to the page:

Reviewer 3: ```suggestion
*Example*: A view that updates the deep linking parameters based on a parameter given in a public API.
The view also implements [interfacename]`HasUrlParameter` to deserialize the parameter in case a user arrives at the page directly:
```
